go/integration_test/processor/calls_test.go

```diff
@@ -396,6 +397,66 @@ func TestProcessor_RecursiveCallsAfterAStaticCallAreStatic(t *testing.T) {
 	}
 }
 
+func TestProcessor_CallingNonExistentAccountFails(t *testing.T) {
+	tests := map[string]struct {
+		account            Account
+		callValue          tosca.Value
+		accountExistsAfter bool
+	}{
+		"empty": {
+			account:            Account{},
+			callValue:          tosca.Value{},
+			accountExistsAfter: false,
+		},
+		"callWithValue": {
+			account:            Account{},
+			callValue:          tosca.NewValue(42),
+			accountExistsAfter: true,
+		},
+		"existingAccount": {
+			account:            Account{Nonce: 1},
+			callValue:          tosca.Value{},
+			accountExistsAfter: true,
+		},
+	}
+	for processorName, processor := range getProcessors() {
+		if strings.Contains(processorName, "floria") {
+			continue
+		}
+		for testName, test := range tests {
+			t.Run(processorName+"/"+testName, func(t *testing.T) {
+				sender := tosca.Address{1}
+				receiver := tosca.Address{2}
+
+				state := WorldState{
+					sender:   Account{Balance: tosca.NewValue(100)},
+					receiver: test.account,
+				}
+				transaction := tosca.Transaction{
+					Sender:    sender,
+					Recipient: &receiver,
+					GasLimit:  sufficientGas,
+					Value:     test.callValue,
+				}
+
+				transactionContext := newScenarioContext(state)
+
+				// Run the processor
+				result, err := processor.Run(tosca.BlockParameters{}, transaction, transactionContext)
+				if err != nil {
+					t.Errorf("execution failed with error %v", err)
+				}
+				if !result.Success {
+					t.Errorf("execution was not successful")
+				}
+				if test.accountExistsAfter && !transactionContext.AccountExists(receiver) {
```

Comment:
else case should also test that the account that should not have been created also does not exists ?

Reply:
A stricter version of this would be
```suggestion
				if test.accountExistsAfter != transactionContext.AccountExists(receiver) {
```